src/asm/rpn.c

```diff
@@ -56,7 +56,8 @@ static uint8_t *reserveSpace(struct Expression *expr, uint32_t size)
 		else if (expr->nRPNCapacity > MAXRPNLEN / 2)
 			expr->nRPNCapacity = MAXRPNLEN;
 		else
-			expr->nRPNCapacity *= 2;
+			while (expr->nRPNCapacity - expr->nRPNLength < size)
```

Comment:
This introduces the possibility of creating a RPN buffer larger than `MAXRPNLEN`, though.

Reply:
Thanks, I restructured my fix. I also noticed a related crash where the initial allocation of 256 would be insufficient, so that's fixed too.